packages/db/postgres/functions.sql

```diff
@@ -250,6 +250,95 @@ BEGIN
 END
 $$;
 
+DO 
+$$
+BEGIN
+  IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'used_storage') THEN
+    CREATE TYPE used_storage AS (uploaded TEXT, pinned TEXT, total TEXT);
+  END IF;
+END
+$$;
+
+CREATE OR REPLACE FUNCTION user_used_storage(query_user_id BIGINT) 
+RETURNS used_storage
+LANGUAGE plpgsql
+AS
+$$
+DECLARE 
+  used_storage  used_storage;
+  uploaded      BIGINT;
+  pinned        BIGINT;
+  total         BIGINT;
+BEGIN
+  uploaded := 
+    (
+      SELECT COALESCE(SUM(c.dag_size), 0)
+      FROM upload u
+      JOIN content c ON c.cid = u.content_cid
+      WHERE u.user_id = query_user_id::BIGINT
+      AND u.deleted_at is null
+    );
+
+  pinned := 
+    (
+      SELECT COALESCE((
+        SELECT COALESCE(SUM(c.dag_size), 0)
+        FROM psa_pin_request psa_pr
+        JOIN content c ON c.cid = psa_pr.content_cid
+        JOIN pin p ON p.content_cid = psa_pr.content_cid
+        JOIN auth_key a ON a.id = psa_pr.auth_key_id
+        WHERE a.user_id = query_user_id::BIGINT
+        AND psa_pr.deleted_at is null
+        AND p.status = 'Pinned'
```

Comment:
I'm sure you're aware, but adding just as a reminder.
This suffers from the problem highlighted by Oli in this [comment](https://github.com/web3-storage/web3.storage/pull/1044#discussion_r834375035) which you already fix and that we'll be soon in master.

Reply:
Good to have a reminder, have merged from main.